DOC-713: New TableModified event (#1757)
tinymce/tinymce-docs: plugins/table.md

## Events

| Name | Data | Description |
|------| ---- | ----------- |
| newrow | N/A | Fired when a row is created |
| newcell | N/A | Fired when a cell is created |
| ObjectResizeStart | N/A | Fired when a resize action is started on a table, row, column or cell using the resize bars |
| ObjectResized | N/A | Fired when a resize action is finished on a table, row, column or cell using the resize bars |

> | tablemodified | `structure: bool; style: bool` | Fired whenever a table is modified in any way. The `style` and `structure` parameters describe whether the table style, structure, or both were modified. |

Reviewer: ```suggestion
| TableModified | `{ structure?: boolean; style?: boolean }` | Fired whenever style or structural changes are made to a table. The `style` and `structure` parameters describe whether the table style, structure, or both were modified. |
```

Reviewer: Note: I made `structure` and `style` optional in the above, as we aren't always passing them through atm.

Author: I added "The **optional** `style` and `structure` parameters..." too

> | TableModified | `{ structure?: boolean, style?: boolean }` | Fired whenever style or structural changes are made to a table. The optional `style` and `structure` parameters describe whether the table style, structure, or both were modified. |

Reviewer 2: Just wanted to check/verify if we need to mention that `table` is also available or is that something we don't want to document?

> | TableModified | `{ structure?: boolean, style?: boolean }` | Fired whenever style or structural changes are made to a table (included in the event). The optional `style` and `structure` parameters describe whether the table style, structure, or both were modified. |

Reviewer 3: ```suggestion
| TableModified | `{ structure?: boolean, style?: boolean }` | Fired when style or structural changes are made to a table. The optional `style` and `structure` parameters describe whether the table style, structure, or both were modified. |
```
